src/lib.rs

```diff
@@ -0,0 +1,71 @@
+#![cfg_attr(not(test), no_std)]
+
+//! Gimlet Inspector protocol definition.
+//!
+//! The types in this file are intended to be encoded with `hubpack` and stuffed
+//! into a UDP packet. Messages to Gimlet use the `Request` type. Responses from
+//! Gimlet use the `*Response` types -- a response structure may be specific to
+//! the request. In both cases, certain requests/responses may append binary
+//! data _after_ the `hubpack`-encoded data. This is documented below on the
+//! specific items.
+
+use hubpack::SerializedSize;
+use serde::{Serialize, Deserialize};
+
+/// Request format to the inspector agent.
+///
+/// This is an enum so that we implicitly get a protocol version field at the
+/// start of all encoded packets. In the event that multiple versions coexist at
+/// one time, we can add multiple variants. As versions are obsoleted, we can
+/// replace them with stub variants to ensure their indexes don't get reused.
+///
+/// The downside of this approach is that it will fail at 256 versions.
```

Comment:
Is the idea that we could add more items to the `Query` enum, as long as we don't change existing variants?  Either way, it would be good to spell out explicitly.  For what it's worth, MGS allows this as a bump of the CURRENT field (but not the MIN).

Along the same lines, do we care about distinguishing between "I **should** be able to deserialize this, and could not" and "I **should not** be able to deserialize this, and could not"?

Reply:
Given the expected application of the tool, I don't think distinguishing between those cases is useful -- but I think I will add a way of indicating a generic "I didn't catch that" message back, for better user feedback.

And, yeah, what you described about maintaining the `Query` enum is indeed what I expected, I will spell it out.